client/components/domains/use-your-domain-step/index.tsx

```diff
@@ -0,0 +1,340 @@
+import {
+	isDomainBundledWithPlan,
+	isDomainMappingFree,
+	isNextDomainFree,
+} from 'calypso/lib/cart-values/cart-items';
+import {
+	getDomainPrice,
+	getDomainProductSlug,
+	getDomainTransferSalePrice,
+} from 'calypso/lib/domains';
+import useCartKey from 'calypso/my-sites/checkout/use-cart-key';
+import { useSelector } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import {
+	DOMAINS_WITH_PLANS_ONLY,
+	NON_PRIMARY_DOMAINS_TO_FREE_USERS,
+} from 'calypso/state/current-user/constants';
+import { currentUserHasFlag } from 'calypso/state/current-user/selectors';
+import { getProductsList } from 'calypso/state/products-list/selectors';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+type OptionContentProps = {
+	image: string;
+	title: string;
+	reasons: React.ReactNode[];
+	onClick: React.MouseEventHandler;
+	buttonText: string;
+	isPrimary: boolean;
+	isSubmitting: boolean;
+	learnMore: React.ReactNode;
+};
+
+function UseYourDomainStepContent( {
+	image,
+	title,
+	reasons,
+	onClick,
+	buttonText,
+	isPrimary,
+	isSubmitting,
+	learnMore,
+}: OptionContentProps ) {
+	return (
+		<Card className="use-your-domain-step__option" compact>
+			<div className="use-your-domain-step__option-inner-wrap">
+				<div className="use-your-domain-step__option-content">
+					<div className="use-your-domain-step__option-illustration">
+						<img src={ image } alt="" />
+					</div>
+					<h3 className="use-your-domain-step__option-title">{ title }</h3>;
+					<div className="use-your-domain-step__option-reasons">
+						{ reasons.map( ( phrase, index ) => {
+							if ( isEmpty( phrase ) ) {
+								return;
+							}
+
+							return (
+								<div className="use-your-domain-step__option-reason" key={ index }>
+									<Gridicon icon="checkmark" size={ 18 } />
+									{ phrase }
+								</div>
+							);
+						} ) }
+					</div>
+				</div>
+				<div className="use-your-domain-step__option-action">
+					<Button
+						className="use-your-domain-step__option-button"
+						primary={ isPrimary }
+						onClick={ onClick }
+						busy={ isSubmitting }
+					>
+						{ buttonText }
+					</Button>
+					<div className="use-your-domain-step__learn-more">{ learnMore }</div>
+				</div>
+			</div>
+		</Card>
+	);
+}
+
+type UseYourDomainStepProps = {
+	analyticsSection?: string;
+	basePath: string;
+	goBack?: () => void;
+	initialQuery?: string;
+	isSignupStep?: boolean;
+	mapDomainUrl?: string;
+	transferDomainUrl?: string;
+	forcePrecheck?: boolean;
+};
+
+function UseYourDomainStep( {
+	analyticsSection = 'domains',
+	basePath,
+	goBack,
+	initialQuery,
+	isSignupStep,
+	mapDomainUrl,
+	transferDomainUrl,
+	forcePrecheck = false,
+}: UseYourDomainStepProps ) {
+	const cartKey = useCartKey();
+	const { responseCart: cart } = useShoppingCart( cartKey );
```

Comment:
The `withShoppingCart` HOC that was used previously called the `useManagerClient` hook, which doesn't seem to be exported from the `@automattic/shopping-cart` package:

https://github.com/Automattic/wp-calypso/blob/6e6636e9633456a68032d28fafd6d734e08db379/packages/shopping-cart/src/with-shopping-cart.tsx#L17-L19

@sirbrillig Is it necessary to call `useManagerClient` here as well, or can we do without and just call `useShoppingCart` ?

Reply:
As the comment says, that `useManagerClient` call is there only to throw a better error if you fail to render `ShoppingCartProvider` as a parent component. It's either:
```
withShoppingCart must be used inside a ShoppingCartProvider
```
or
```
useShoppingCart must be used inside a ShoppingCartProvider
```
Note that the error message matches the API that you really used.

Now when you're using `useShoppingCart` directly, the extra `useManagerClient` would have no function.